src/test/ui/self/elision/ref-struct-async.nll.stderr

```diff
@@ -47,6 +67,11 @@ LL |     async fn box_pin_Struct(self: Box<Pin<&Struct>>, f: &u32) -> &u32 {
    |                                           let's call the lifetime of this reference `'2`
 LL |         f
    |         ^ associated function was supposed to return data with lifetime `'2` but it is returning data with lifetime `'1`
+   |
+help: consider introducing a named lifetime parameter and update trait if needed
+   |
+LL |     async fn box_pin_Struct<'a>(self: Box<Pin<&'a Struct>>, f: &'a u32) -> &u32 {
+   |                            ++++                ++               ++
```

Comment:
Should the suggestion be also pointing at the return type and suggesting `-> &'a u32`?

I'm guessing this is because of the desugared result of the `fn` hiding the return type, which means it would have to be special cased somehow.

Reply:
Ah! Ignore me, this is the *current* output of that method, disregard.